### PR TITLE
[SPARK-12603] [MLlib] PySpark MLlib GaussianMixtureModel should support single instance predict/predictSoft

### PR DESCRIPTION
**Pull request by yanboliang**
PySpark MLlib `GaussianMixtureModel` should support single instance `predict/predictSoft` just like Scala do.

### CODE REVIEW

**Comment by jkbradley** [comment]
These changes look fine to me, but could you please test predictSoft in Python (also to add this to the example in the doc)?

File: examples/src/main/scala/org/apache/spark/examples/mllib/DenseGaussianMixture.scala [hunk 1]
```diff
@@ -58,6 +58,11 @@ object DenseGaussianMixture {
         (clusters.weights(i), clusters.gaussians(i).mu, clusters.gaussians(i).sigma))
     }
 
+    println("The membership value of each vector to all mixture components (first <= 100):")
+    val membership = clusters.predictSoft(data)
+    membership.take(100).foreach { x =>
+      print(" " + x.mkString(","))
+    }
```
Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/DenseGaussianMixture.scala]: I'd add a println after this for spacing.

**Comment by jkbradley** [comment]
Looks good except for the 1 comment

**Comment by jkbradley** [comment]
I'm finding that this unit test fails on my laptop.  I'm wondering if the problem is: different default parallelism levels on different machines => clusterdata_1 has different numbers of partitions => GMM gets initialized with different samples.

Could you please change the sc.parallelize call to use 2 partitions to see if that fixes the test?

**Comment by yanboliang** [comment]
@jkbradley It can produce stable result after changing ```sc.parallelize``` to use 2 partitions. Thanks! 
Due to the ```predictSoft``` output may be negligibly small that we can not use ```0.000...``` to match equality, so I use ```abs(softPredicted[1] - 0.0) < 0.001``` in doctest.

**Comment by jkbradley** [comment]
That looks good.  I also just noticed: Could you please update the docs for Python predict, predictSoft to say they work on RDDs and single vectors?  That should be it.

**Comment by jkbradley** [comment]
LGTM
Merging with master
Thanks for the PR!

**Pull request closed without merging**